src/analytics/src/main/scala/osmesa/analytics/oneoffs/ChangesetStreamProcessor.scala

```diff
@@ -183,7 +205,11 @@ object ChangesetStreamProcessor extends CommandApp(
               this.partitionId = partitionId
               this.version = version
 
-              connection = DriverManager.getConnection(s"jdbc:${databaseUri.toString}")
+              connection = (databaseUser, databasePass).mapN { (usr, pass) =>
+                DriverManager.getConnection(s"jdbc:${databaseUri.toString}", usr, pass)
```

Comment:
Username and password can be included in a JDBC URI as `?user=<username>&password=<password>`, which can simplify this a bit.

Ideally we'd take username and password from the auth part of a URI and copy it in there; that makes it easy to extract from an environment variable: `... --database-uri $DATABASE_URL`

Reply:
I see that the URI currently used is `postgresql://[host]/[db]?user=[username]&password=[password]` which doesn't square with `postgresql://[user[:password]@][netloc][:port][/dbname][?param1=value1&...]` format mentioned https://www.postgresql.org/docs/9.3/static/libpq-connect.html

We can/should iterate through these, but wondering why the diff.

Reply:
You and me both. It looks like a JDBC wart where auth is ignored when passed in the typical part of a URI.